cicecore/cicedynB/dynamics/ice_dyn_shared.F90

```diff
@@ -605,13 +606,17 @@ subroutine dyn_prep2 (nx_block,   ny_block,   &
       !-----------------------------------------------------------------
 
       icellu = 0
+         
       do j = jlo, jhi
       do i = ilo, ihi
-
-         ! ice extent mask (U-cells)
          iceumask_old(i,j) = iceumask(i,j) ! save
-         iceumask(i,j) = (umask(i,j)) .and. (aiu  (i,j) > a_min) & 
-                                      .and. (umass(i,j) > m_min)
+         if (grid_system == 'B') then ! include ice mask.
+         ! ice extent mask (U-cells)
+            iceumask(i,j) = (umask(i,j)) .and. (aiu  (i,j) > a_min) & 
+                                         .and. (umass(i,j) > m_min)
+         else  ! ice mask shpuld be applied to cd grid. For now it is not implemented.
```

Comment:
I think we agreed on spelling out `elseif ( trim(grid_sytem) == 'CD')` (or using a `select` statement instead). It's also clearer when we read the code, I think.

Reply:
Obsolete. see soon upcoming push  that make cd grid run complete